okio/src/unixMain/kotlin/okio/-UnixPosixVariant.kt

```diff
@@ -113,6 +114,22 @@ internal actual fun PosixFileSystem.variantOpenReadWrite(file: Path): FileHandle
   return UnixFileHandle(true, openFile)
 }
 
+@ExperimentalFileSystem
+internal actual fun PosixFileSystem.variantCreateSymlink(source: Path, target: Path) {
+  if (source.parent == null || !exists(source.parent!!)) {
+    throw IOException("parent directory does not exist: ${source.parent}")
+  }
+
+  if (exists(source)) {
+    throw IOException("already exists: $source")
+  }
+
+  val result = symlink(target.toString(), source.toString())
```

Comment:
I cannot use canonicalize (as we're doing in FakeFS) because this method throws if the file doesn't exist.
Tests seem to pass. Could that be a problem?

Reply:
I think this looks right as-is.

Reply:
Also FakeFileSystem isn’t calling `canonicalize()`, it’s merely converting a potentially-relative path to an absolute path and calling that canonical.